lnwallet/interface.go

```diff
@@ -594,6 +596,67 @@ type MessageSigner interface {
 		doubleHash bool) (*ecdsa.Signature, error)
 }
 
+// AddrWithKey wraps a normal addr, but also includes the internal key for the
+// delivery addr if known.
+type AddrWithKey struct {
+	lnwire.DeliveryAddress
+
+	InternalKey fn.Option[btcec.PublicKey]
+
+	// TODO(roasbeef): consolidate w/ instance in chan closer
+}
+
+// InternalKeyForAddr returns the internal key associated with a taproot
+// address.
+func InternalKeyForAddr(wallet WalletController, netParams *chaincfg.Params,
+	deliveryScript []byte) (fn.Option[keychain.KeyDescriptor], error) {
+
+	none := fn.None[keychain.KeyDescriptor]()
+
+	// If it's not a taproot address, we don't require to know the internal
+	// key in the first place. So we don't return an error here, but also no
+	// internal key.
+	if !txscript.IsPayToTaproot(deliveryScript) {
```

Comment:
IIRC we did initially, but then that needed special handling at the caller side. In this case, we return `None`, as there just isn't an internal key for non taproot addrs.

Reply:
Yeah, we call this user defined delivery addresses (e.g. upfront shutdown address). So we can't error out, as that would break the unit and integration tests.
I think we'll eventually need to find a workaround if we want to support upfront shutdown addresses for Taproot Asset Channels...